pkg/gardenlet/controller/shoot/care/reconciler.go

```diff
@@ -59,16 +61,46 @@ type Reconciler struct {
 }
 
 // Reconcile executes care operations, e.g. health checks or garbage collection.
-func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+func (r *Reconciler) Reconcile(ctx context.Context, req Request) (reconcile.Result, error) {
 	log := logf.FromContext(ctx)
 
 	// Timeout for all calls (e.g. status updates), give status updates a bit of headroom if health checks
 	// themselves run into timeouts, so that we will still update the status with that timeout error.
 	ctx, cancel := controllerutils.GetMainReconciliationContext(ctx, r.Config.Controllers.ShootCare.SyncPeriod.Duration)
 	defer cancel()
 
+	if !req.IsManagedResource {
+		return r.reconcileShoot(ctx, log, req.NamespacedName)
+	}
+
+	shoots := &gardencorev1beta1.ShootList{}
+	if err := r.GardenClient.List(context.Background(), shoots, client.MatchingFields{
+		core.ShootStatusTechnicalID: req.Namespace,
+	}); err != nil {
+		return reconcile.Result{}, fmt.Errorf("error looking up shoot by technical id: %w", err)
+	}
+
+	if len(shoots.Items) == 0 {
+		log.V(1).Info("No shoot found for managed resource, ignoring it")
+		return reconcile.Result{}, nil
+	} else {
+		result, err := r.reconcileShoot(ctx, log, types.NamespacedName{
+			Name:      shoots.Items[0].Name,
+			Namespace: shoots.Items[0].Namespace,
+		})
+		if err != nil {
+			return result, err
+		}
+		// Reconciles triggered due to a changed managed resource are one time checks.
+		// Periodic checks are already performed based on the existing shoot objects.
+		return reconcile.Result{}, nil
```

Comment:
Can this be simplified to 
```suggestion
		return r.reconcileShoot(ctx, log, types.NamespacedName{
			Name:      shoots.Items[0].Name,
			Namespace: shoots.Items[0].Namespace,
		})
```

?

Reply:
Not exactly. Right now, there are two request values that can cause a reconcile of the underlying shoot. There's the already existing periodic shoot object reconcile. And there are the `Request` objects triggered by a ManagedResource change. Those should only be processed once and reconcile themselves again.

I'm starting to wonder whether it would be better to perform the ManagedResource to Shoot mapping in the `EnqueueRequestsFromMapFunc` even though that would mean to drop the ManagedResource on the floor in case of an error and just log an error. That might also make it easier to properly rate limit the shoot reconcile.